controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,293 @@
+		odfInfoKeyName := obj.getOdfInfoKeyName(storageCluster)
+		delete(odfInfoConfigMap.Data, odfInfoKeyName)
+		if err = r.Client.Update(r.ctx, odfInfoConfigMap); err != nil {
+			r.Log.Error(err, "Failed to update odf-info config map with deleted key.", "ConfigMap", klog.KRef(odfInfoConfigMap.Namespace, odfInfoConfigMap.Name), "Key", odfInfoKeyName)
+			return reconcile.Result{}, fmt.Errorf("failed to delete key %v in odf-info %v: %v", odfInfoKeyName, odfInfoConfigMap.Name, err)
+		}
+	} else {
+		if err = r.Client.Delete(r.ctx, odfInfoConfigMap); err != nil && !errors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to delete odf-info config map.", "ConfigMap", klog.KRef(odfInfoConfigMap.Namespace, odfInfoConfigMap.Name))
+			return reconcile.Result{}, fmt.Errorf("failed to delete odf-info %v: %v", odfInfoConfigMap.Name, err)
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func getOdfInfoData(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster, clients string) (string, error) {
+	odfVersion, err := getOdfVersion(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+	cephFSId, err := getCephFsid(r, storageCluster)
+	if err != nil {
+		return "", err
+	}
+
+	odfDeploymentType := odfDeploymentTypeExternal
+	isDROptimized := false
+	// Set isDROptmized to "false" in case of external clusters as we currently don't have to way to determine
+	// if external cluster OSDs are using bluestore-rdr
+	if !storageCluster.Spec.ExternalStorage.Enable {
+		odfDeploymentType = odfDeploymentTypeInternal
+		if isDROptimized, err = getIsDROptimized(r, storageCluster); err != nil {
+			return "", err
+
+		}
+	}
+	var storageSystemName string
+	if storageSystemName, err = getStorageSystemName(storageCluster); err != nil {
+		return "", err
+	}
+
+	connectedClients, err := getConnectedClients(r, storageCluster, clients)
+	if err != nil {
+		return "", err
+	}
+
+	data := OdfInfoData{
+		Version:           odfVersion,
+		DeploymentType:    odfDeploymentType,
+		IsDROptimized:     isDROptimized,
+		StorageSystemName: storageSystemName,
+		Clients:           connectedClients,
+		StorageCluster: infoStorageCluster{
+			NamespacedName:          client.ObjectKeyFromObject(storageCluster),
+			StorageProviderEndpoint: storageCluster.Status.StorageProviderEndpoint,
+			CephClusterFSID:         cephFSId,
+		},
+	}
+	yamlData, err := yaml.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+	return string(yamlData), nil
+
+}
+
+func getConnectedClients(r *StorageClusterReconciler, storageCluster *ocsv1.StorageCluster, clients string) ([]connectedClient, error) {
+	storageConsumers := &v1alpha1.StorageConsumerList{}
+	err := r.Client.List(r.ctx, storageConsumers, client.InNamespace(storageCluster.Namespace))
+	if err != nil {
+		return nil, err
+	}
+	var connectedClients []connectedClient
+	if err = yaml.Unmarshal([]byte(clients), connectedClients); err != nil {
+		return nil, err
+	}
+
+	for storageConsumerID := range storageConsumers.Items {
```

Comment:
It is not an id, just an index into the Items array:
```suggestion
	for i := range storageConsumers.Items {
```